### PR TITLE
✨ update: tls.ClientHelloInfo in Ctx

### PR DESCRIPTION
**Pull request by thomasdseao**
This PR is related to a request : https://github.com/orgs/gofiber/projects/1/views/4

Add the function ClientHelloInfo to the fiber Ctx so everybody can access it from handler.

### CODE REVIEW

File: app.go [hunk 1]
```diff
@@ -366,6 +366,9 @@ type Config struct {
 	//
 	// Optional. Default: DefaultColors
 	ColorScheme Colors `json:"color_scheme"`
+
+	// TLS handler
+	tlsHandler *tlsHandler
```
Comment (efectn) [inline app.go:371]: This can be a part of App struct

File: ctx_test.go [hunk 2]
```diff
@@ -1247,6 +1247,28 @@ func Test_Ctx_Method(t *testing.T) {
...
 }
 
+// go test -run Test_Ctx_ClientHelloInfo
+func Test_Ctx_ClientHelloInfo(t *testing.T) {
+	t.Parallel()
+	app := New()
+	app.Post("/test", func(c *Ctx) error {
+		result := c.ClientHelloInfo()
+		if result != nil {
+			panic("Invalid ClientHelloInfo, it should be nil")
```
Comment (efectn) [inline ctx_test.go:1257]: Return an error with NewError instead of `panic()`. And we may need more tests i suppose

File: app.go (hunk 1 above)
Comment (thomasdseao) [reply app.go]: Changed 👍

File: ctx_test.go (hunk 2 above)
Comment (thomasdseao) [reply ctx_test.go]: I add more tests on this, please check it and tell me if it looks good for you 👍

**Comment by efectn** [comment]
I think this method should be in App. Can you move it into the app?

**Comment by wireguard-dev** [comment]
> I think this method should be in App. Can you move it into the app?

@efectn Why? I think it must be in context directly, not in App()

**Comment by efectn** [comment]
> > I think this method should be in App. Can you move it into the app?
> 
> @efectn Why? I think it must be in context directly, not in App()

It doesn't have context-specific thing

**Comment by wireguard-dev** [comment]
> > > I think this method should be in App. Can you move it into the app?
> > 
> > 
> > @efectn Why? I think it must be in context directly, not in App()
> 
> It doesn't have context-specific thing

tls.ClientHelloInfo, is sent by the browser when the user connects to the website, so I think it should be in context.

**Comment by thomasdseao** [comment]
Yes the CHI is sent by the client, that's why I put it in the context.
But if you think this is more compliant with Fiber logical to have it in App, I can move it.

**Comment by ReneWerner87** [comment]
@thomasdseao 
can you please add a pull request to our documentation repository so that everyone knows what the point of it is and how to use it?
https://github.com/gofiber/docs/pulls

**Comment by thomasdseao** [comment]
@ReneWerner87 That's done : https://github.com/gofiber/docs/pull/271

**Review by ReneWerner87 — APPROVED** [review]

File: ctx.go [hunk 3]
```diff
@@ -130,6 +143,8 @@ func (app *App) AcquireCtx(fctx *fasthttp.RequestCtx) *Ctx {
 	c.fasthttp = fctx
 	// reset base uri
 	c.baseURI = ""
+	// Attach tlsHandler object to context
+	c.tlsHandler = app.tlsHandler
```
Comment (efectn) [inline ctx.go:147]: i think we can get it as `c.app.tlsHandler` instead of adding new field.

**Pull request merged**